### PR TITLE
Fix issue 573

### PR DESCRIPTION
**Pull request by jcp19**

### CODE REVIEW

File: src/main/scala/viper/gobra/frontend/info/implementation/resolution/Enclosing.scala [hunk 1]
```diff
@@ -72,6 +72,12 @@ trait Enclosing { this: TypeInfoImpl =>
   lazy val tryEnclosingFunction: PNode => Option[PFunctionDecl] =
     down[Option[PFunctionDecl]](None) { case m: PFunctionDecl => Some(m) }
 
+  lazy val tryEnclosingFunctionOrMethod: PNode => Option[PFunctionOrMethodDecl] =
+    down[Option[PFunctionOrMethodDecl]](None) {
+      case f: PFunctionDecl => Some(f)
+      case m: PMethodDecl => Some(m)
```
Comment (ArquintL) [inline src/main/scala/viper/gobra/frontend/info/implementation/resolution/Enclosing.scala:78]: Wouldn't it make more sense to match here on PFunctionOrMethodDecl?

**Review by ArquintL — APPROVED** [review]
One small comment otherwise looks good to me

File: src/main/scala/viper/gobra/frontend/info/implementation/resolution/Enclosing.scala (hunk 1 above)
Comment (jcp19) [reply src/main/scala/viper/gobra/frontend/info/implementation/resolution/Enclosing.scala:78]: Good point! I will change it

**Pull request merged**